src/tasks/AndroidAppBuilder/Templates/monodroid-coreclr.c

```diff
@@ -130,9 +129,64 @@ get_tpas_from_path(const char* dir_path, const char** tpas)
 }
 
 static int
-mono_droid_runtime_init (const char* executable, int managed_argc, char* managed_argv[], int local_date_time_offset)
+bundle_executable_path (const char* executable, const char* bundle_path, const char** executable_path)
 {
-    LOG_INFO ("mono_droid_runtime_init called with executable: %s", executable);
+    size_t executable_path_len = strlen(bundle_path) + strlen(executable) + 1; // +1 for '/'
+    char* temp_path = (char*)malloc(sizeof(char) * executable_path_len + 1); // +1 for '\0'
+    if (temp_path == NULL)
+    {
+        return -1;
+    }
+
+    size_t res = snprintf(temp_path, (executable_path_len + 1), "%s/%s", bundle_path, executable);
+    if (res < 0 || res != executable_path_len)
+    {
+        return -1;
+    }
+    *executable_path = temp_path;
+    return executable_path_len;
+}
+
+static void
+free_resources ()
+{
+    if (g_bundle_path)
+    {
+        free (g_bundle_path);
+        g_bundle_path = NULL;
+    }
+    if (g_executable_path)
+    {
+        free (g_executable_path);
+        g_executable_path = NULL;
+    }
+    if (g_coreclr_handle)
+    {
+        coreclr_shutdown (g_coreclr_handle, g_coreclr_domainId);
+        g_coreclr_handle = NULL;
+    }
```

Comment:
Just an FYI. Calling `coreclr_shutdown` doesn't really make the runtime unloadable. I personally think we should avoid calling this API in general because it isn't like we can reinitialize CoreCLR or unload anything. It is kind of bookkeeping and in general leads to a false sense of "clean state".

You can absolutely leave it in, but it doesn't really do what one might think it does.

Reply:
Good point. My assumption was that this should be called after the workload is finished. If it's not needed I'll remove it for simplicity.

Reply:
Thinking about this, is there a proper way to free the resources hold by `g_coreclr_handle` or is this something we don't need to worry about?

Reply:
This is the only API that releases many of those resources. Releasing them is completely reasonable. My main points with this comment were that (a) this doesn't actually shutdown the runtime in a way that makes it re-activatable or unloadable - that isn't intuitive based on its name, and (b) CoreCLR generally assumes it "owns" the process, in this case it doesn't, and this API will only be called on the way to process termination.

Based on what I'm seeing, this path is only called during the process termination sequence which makes it completely reasonable to keep. My suggestion would simply be to add a comment such as:

> Clean up some coreclr resources. This doesn't make coreclr unloadable.